geo/src/algorithm/frechet_distance.rs

```diff
@@ -72,19 +72,26 @@ impl<'a, T> Data<'a, T>
 where
     T: GeoFloat + FromPrimitive,
 {
-    fn compute(&mut self, i: usize, j: usize) -> T {
-        if self.cache[i][j].is_nan() {
-            let eucl = Point::from(self.ls_a[i]).euclidean_distance(&Point::from(self.ls_b[j]));
-            self.cache[i][j] = match (i, j) {
-                (0, 0) => eucl,
-                (_, 0) => self.compute(i - 1, 0).max(eucl),
-                (0, _) => self.compute(0, j - 1).max(eucl),
-                (_, _) => ((self.compute(i - 1, j).min(self.compute(i - 1, j - 1)))
-                    .min(self.compute(i, j - 1)))
-                .max(eucl),
-            };
```

Comment:
Please leave a comment pointing to the source of this implementation (https://github.com/joaofig/discrete-frechet/tree/master? Or a more direct link), and a brief explanation of how it works.

Reply:
The implementation is not exactly the same, as I use a single vector instead of a matrix to reduce number of allocations and improve data locality, but I will add a link to it if you prefer. I did leave all the other links that point to the resources that explain the theory behind it.

Reply:
I have added a link to the reference implementation, let me know if that looks good to you.